fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/mixin/entity/event/ServerPlayerEntityMixin.java

```diff
@@ -90,6 +95,21 @@ private void onCopyFrom(ServerPlayerEntity oldPlayer, boolean alive, CallbackInf
 		ServerPlayerEvents.COPY_FROM.invoker().copyFromPlayer(oldPlayer, (ServerPlayerEntity) (Object) this, alive);
 	}
 
+    @Inject(method = "teleport(Lnet/minecraft/server/world/ServerWorld;DDDLjava/util/Set;FF)Z", at = @At("HEAD"), cancellable = true)
+    private void beforeTeleport(ServerWorld world, double destX, double destY, double destZ, Set<PositionFlag> flags, float yaw, float pitch, CallbackInfoReturnable<Boolean> cir) throws CommandSyntaxException {
+        ServerPlayerEntity player = (ServerPlayerEntity) (Object) this;
+        boolean allowed = ServerPlayerEvents.ALLOW_TELEPORT.invoker().allowTeleport(player, new Vec3d(destX, destY, destZ));
+        if (!allowed) {
+            throw new CommandSyntaxException(null, new LiteralMessage(""));
```

Comment:
Question (I havent got the code in front of me), why is a CommandSyntaxException thrown? Does it need a message, and does this prevent the player from being teleported when they use a portal?

Reply:
I threw an exception because, whether the teleportation is successful or not, a teleportation message will always pop up when using coordinates or player-to-player teleportation. An empty exception can somewhat resolve this issue.